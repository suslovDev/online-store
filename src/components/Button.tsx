import { FC } from "react";
import { IButton } from "./IButton";
import classes from "./Button.module.css";

const Button: FC<IButton> = ({ variant, onClick, children }) => {
  return (
    <>
      {variant === "buy" && (
        <button className={classes.button} onClick={onClick}>
          {children}
          <img src='./buy-img.png' width={27} alt='' />
        </button>
      )}
      {variant === "catalog " && (
        <button
          className={`${classes.button} ${classes.standart}`}
          onClick={onClick}>
          {children}
          <img src='./catalog-img.png' width={27} alt='' />
        </button>
      )}
      {variant === "search" && (
        <button
          className={`${classes.button} ${classes.round}`}
          onClick={onClick}>
          {children}
          <img src='./search-img.png' width={18} alt='' />
        </button>
      )}
      {variant === "download" && (
        <button
          className={`${classes.button} ${classes.large}`}
          onClick={onClick}>
          {children}
          <img src='./dwld-img.png' width={27} alt='' />
        </button>
      )}
      {variant === "remove" && (
        <button
          className={`${classes.button} ${classes["round-large"]}`}
          onClick={onClick}>
          {children}
          <img src='./remove-img.png' width={27} alt='' />
        </button>
      )}
    </>
  );
};

export default Button;
