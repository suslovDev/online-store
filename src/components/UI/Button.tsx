import { FC } from "react";
import { IButton } from "./IButton";
import classes from "./Button.module.css";

const Button: FC<IButton> = ({ variant, type, onClick, children }) => {
  return (
    <>
      {variant === "standart" && (
        <button
          type={type}
          className={`${classes.button} ${classes.large}`}
          onClick={onClick}>
          {children}
        </button>
      )}
      {variant === "buy" && (
        <button type={type} className={classes.button} onClick={onClick}>
          {children}
          <img src='./buy-img.png' width={27} alt='' />
        </button>
      )}
      {variant === "buy-large" && (
        <button
          type={type}
          className={`${classes.button} ${classes.large}`}
          onClick={onClick}>
          {children}
          <img src='./buy-img.png' width={27} alt='' />
        </button>
      )}
      {variant === "catalog " && (
        <button
          type={type}
          className={`${classes.button} ${classes.standart}`}
          onClick={onClick}>
          {children}
          <img src='./catalog-img.png' width={15} alt='' />
        </button>
      )}
      {variant === "search" && (
        <button
          type={type}
          className={`${classes.button} ${classes.round}`}
          onClick={onClick}>
          {children}
          <img src='./search-img.png' width={18} alt='' />
        </button>
      )}
      {variant === "download" && (
        <button
          type={type}
          className={`${classes.button} ${classes.large}`}
          onClick={onClick}>
          {children}
          <img src='./dwld-img.png' width={17} alt='' />
        </button>
      )}
      {variant === "remove" && (
        <button
          type={type}
          className={`${classes.button} ${classes["round-large"]}`}
          onClick={onClick}>
          {children}
          <img src='./remove-img.png' width={27} alt='' />
        </button>
      )}
    </>
  );
};

export default Button;
