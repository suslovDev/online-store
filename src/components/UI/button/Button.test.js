import { cleanup } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Button from "./Button";
afterEach(cleanup);
describe("Button.jsx:", () => {
  it("Выводится корректно с variant: 'standart'", () => {
    const tree = TestRenderer.create(<Button variant='standart' />);
    expect(tree).toMatchSnapshot();
  });
  it("Выводится корректно с variant: 'buy'", () => {
    const tree = TestRenderer.create(<Button variant='buy' />);
    expect(tree).toMatchSnapshot();
  });
  it("Выводится корректно с variant: 'buy-large'", () => {
    const tree = TestRenderer.create(<Button variant='buy-large' />);
    expect(tree).toMatchSnapshot();
  });
  it("Выводится корректно с variant: 'search'", () => {
    const tree = TestRenderer.create(<Button variant='search' />);
    expect(tree).toMatchSnapshot();
  });
  it("Выводится корректно с variant: 'download'", () => {
    const tree = TestRenderer.create(<Button variant='download' />);
    expect(tree).toMatchSnapshot();
  });
  it("Выводится корректно с variant: 'remove'", () => {
    const tree = TestRenderer.create(<Button variant='remove' />);
    expect(tree).toMatchSnapshot();
  });
  it("Выводится корректно с variant: 'catalog'", () => {
    const tree = TestRenderer.create(<Button variant='catalog' />);
    expect(tree).toMatchSnapshot();
  });
});
