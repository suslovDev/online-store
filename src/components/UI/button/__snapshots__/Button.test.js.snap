// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.jsx: Выводится корректно с variant: 'buy' 1`] = `
<button
  className="button"
>
  <img
    alt=""
    src="./buy-img.png"
    width={27}
  />
</button>
`;

exports[`Button.jsx: Выводится корректно с variant: 'buy-large' 1`] = `
<button
  className="button large"
>
  <img
    alt=""
    src="./buy-img.png"
    width={27}
  />
</button>
`;

exports[`Button.jsx: Выводится корректно с variant: 'catalog' 1`] = `null`;

exports[`Button.jsx: Выводится корректно с variant: 'download' 1`] = `
<button
  className="button large"
>
  <img
    alt=""
    src="./dwld-img.png"
    width={17}
  />
</button>
`;

exports[`Button.jsx: Выводится корректно с variant: 'remove' 1`] = `
<button
  className="button round-large"
>
  <img
    alt=""
    src="./remove-img.png"
    width={27}
  />
</button>
`;

exports[`Button.jsx: Выводится корректно с variant: 'search' 1`] = `
<button
  className="button round"
>
  <img
    alt=""
    src="./search-img.png"
    width={18}
  />
</button>
`;

exports[`Button.jsx: Выводится корректно с variant: 'standart' 1`] = `
<button
  className="button large"
/>
`;
